### PR TITLE
Stress: Enable per driver loader and container option overloading

### PR DESCRIPTION
**Pull request by anthony-murphy**
This change enables testing client provided protocol tree which is supported by all services.

Related to #8080

### CODE REVIEW

**Comment by vikagrawal** [comment]
Please add spacing before/after { }  => for styling consistency.

Also, override is added only for two profiles. Are they applicable to other profiles?

File: packages/test/test-service-load/src/optionsMatrix.ts [hunk 1]
```diff
@@ -25,10 +25,29 @@ const loaderOptionsMatrix: OptionsMatrix<ILoaderOptions> = {
...
+export function applyOverrides<T>(options: OptionsMatrix<T>, optionsOverrides: Partial<OptionsMatrix<T>> | undefined) {
+    const realOptions: OptionsMatrix<T> = {...options};
+    if(optionsOverrides !== undefined) {
+        for(const key of Object.keys(optionsOverrides)) {
+            const override = optionsOverrides[key];
+            if(override !== undefined) {
+                if(Array.isArray(override)) {
+                    realOptions[key] =  override;
+                }else{
+                    console.error(`Override for ${key} is not array: ${JSON.stringify(optionsOverrides)}`);
```
Comment (vladsud) [inline packages/test/test-service-load/src/optionsMatrix.ts:37]: I'd prefer fail fast (exception) over console logs

**Review by vladsud — APPROVED** [review]

File: packages/test/test-service-load/src/optionsMatrix.ts [hunk 2]
```diff
@@ -31,10 +31,10 @@ export function applyOverrides<T>(options: OptionsMatrix<T>, optionsOverrides: P
         for(const key of Object.keys(optionsOverrides)) {
             const override = optionsOverrides[key];
             if(override !== undefined) {
-                if(Array.isArray(override)) {
+                if(!Array.isArray(override)) {
```
Comment (vikagrawal) [inline packages/test/test-service-load/src/optionsMatrix.ts:34]: Forgot to remove not(!)

File: packages/test/test-service-load/src/optionsMatrix.ts [hunk 3]
```diff
@@ -25,10 +25,29 @@ const loaderOptionsMatrix: OptionsMatrix<ILoaderOptions> = {
...
-export const generateLoaderOptions = (seed: number): ILoaderOptions[]=>
-    generatePairwiseOptions<ILoaderOptions>(
-        loaderOptionsMatrix,
+export function applyOverrides<T>(options: OptionsMatrix<T>, optionsOverrides: Partial<OptionsMatrix<T>> | undefined) {
+    const realOptions: OptionsMatrix<T> = {...options};
+    if(optionsOverrides !== undefined) {
+        for(const key of Object.keys(optionsOverrides)) {
+            const override = optionsOverrides[key];
+            if(override !== undefined) {
+                if(!Array.isArray(override)) {
```
Comment (anthony-murphy) [inline packages/test/test-service-load/src/optionsMatrix.ts:34]:
```suggestion
                if(Array.isArray(override)) {
```

**Comment by anthony-murphy** [comment]
> @anthony-murphy Please add spacing before/after { } => for styling consistency.
> 
> Are overrides here for testing only? What's impact of those override? Want to understand should we use override for scale testing too.

The impact of those overrides enable #5771. i will want to run scale test with it enabled (started offline thread with you) to work out the details there

**Review by vikagrawal — APPROVED** [review]

**Pull request merged**